services/api/src/resources/project/harborSetup.ts

```diff
@@ -50,8 +50,31 @@ async function createHarborProject(sqlClient: MariaClient, harborClient, lagoonP
 
   // Get new harbor project's id
   try {
-    const res = await harborClient.get(`projects?name=${lagoonProjectName}`)
-    var harborProjectID = res.body[0].project_id
+    // Grab paginated project list results
+    const pageSize = 100
+    let results = []
+    let res = await harborClient.get(`projects?name=${lagoonProjectName}&page_size=${pageSize}`)
+
+    if (parseInt(res.headers['x-total-count']) > pageSize) {
+      let i = 1
+      while (res.body != null) {
+        results = results.concat(res.body)
+        i++
+        res = await harborClient.get(`projects?name=${lagoonProjectName}&page_size=${pageSize}&page=${i}`)
+        console.log(res.body)
```

Comment:
Was this overlooked? That's gonna add a lot of cruft to sift through if anyone needs to look at the API logs.

Reply:
Good catch; removed